jdbc/src/java/org/apache/hive/jdbc/HiveConnection.java

```diff
@@ -357,6 +357,15 @@ public HiveConnection(String uri, Properties info) throws SQLException {
           }
 
           if (numRetries >= maxRetries) {
+            try {
```

Comment:
I was looking around and made the following notices:
* this check only occurs right before we throw the exception out
* what will happen with opened connection in between retries?
* it seems to me that `openTransport` does a check on the transport that is open or not - but it creates a new transport anyway; so I assume the driver will open a fresh connection for every try

I'm thinking the following:
* what do you think about closing the connection after a failed attempt?

Reply:
yes, you're right. Thanks very much for the advice and comments!